scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/session/mcap/FFMPEGVideoDataReader.java

```diff
@@ -55,14 +61,42 @@ public long readFrameAtTimestamp(long timestamp)
             throw new RuntimeException(e);
          }
       }
-      return currentTimestamp.get();
+   }
+
+   public void readCurrentFrame()
+   {
+      try
+      {
+         long clampedTime = Math.min(maxVideoTimestamp, Math.max(0, currentTimestamp.get() + playbackOffset.get()));
+         frameGrabber.setVideoTimestamp(clampedTime);
+         currentFrame = frameGrabber.grabFrame();
```

Comment:
This method should just rely on the other method `readFrameAtTimestamp` pass `currentTimestemp.get()`, unless I'm missing something.

Reply:
Oh yeah that ended up being this way because `currentTimestamp` stores the video timestamp (in the weird FFMpeg microsecond units) but `readFrameAtTimestamp` expects a nanosecond timestamp from the log reader. I think I'll just switch to using the nanosecond timestamp everywhere.

Reply:
Fixed